prottp.go

```diff
@@ -58,6 +59,26 @@ func NewError(statusCode int, body proto.Message) ErrorWithStatus {
 	return &respError{statusCode: statusCode, body: body}
 }
 
+type key int
+
+const headerKey key = iota
+
+func WithHeader(h http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		newCtx := context.WithValue(ctx, headerKey, http.Header{})
```

Comment:
If you use `w.Header()` instead of `http.Header{}`, then there's no need to write headers in `WriteMessage`?